src/Analyzers/Core/Analyzers/RemoveRedundantEquality/RemoveRedundantEqualityDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.RemoveRedundantEquality
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    internal sealed class RemoveRedundantEqualityDiagnosticAnalyzer
+        : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        private static readonly LocalizableResourceString s_title =
+            new(nameof(AnalyzersResources.Remove_redundant_equality), AnalyzersResources.ResourceManager, typeof(AnalyzersResources));
+
+        public RemoveRedundantEqualityDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.RemoveRedundantEqualityWithTrueDiagnosticId,
+                  option: null,
+                  s_title)
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterOperationAction(AnalyzeBinaryOperator, OperationKind.BinaryOperator);
+
+        private void AnalyzeBinaryOperator(OperationAnalysisContext context)
+        {
+            var operation = (IBinaryOperation)context.Operation;
+            if (operation.OperatorMethod is not null) // Overloaded operator
+            {
+                return;
+            }
+
+            if (operation.OperatorKind != BinaryOperatorKind.Equals)
+            {
+                return;
+            }
```

Comment:
```suggestion
            }
            
```

blank line after `}`